src/Service/AnrInstanceService.php

```diff
@@ -1384,7 +1384,9 @@ public function importFromArray($data, $anr, $idParent = null, $modeImport = 'me
                                             'uuid' => $measure_array['referential']]);
                       $soaCategories = $this->get('soaCategoryTable')
                                             ->getEntityByFields(['anr' => $anr->id,
-                                                'label' . $this->getLanguage() => $measure_array['category']]);
+                                                'label' . $this->getLanguage() => $measure_array['category'],
+                                                'referential' => ['anr' => $anr->id,
```

Comment:
accessing properties directly (e.g. ->id) initiates calling magic method __isset. We would wish to avoid usage of the magic and call getters setters directly, but good practice reasons and clarity. One of the explanations in this comment https://www.php.net/manual/en/language.oop5.overloading.php#108040.
I can a bit reformat the styles of this part.